### PR TITLE
Add HFE export functionality

### PR DESCRIPTION
**Pull request by mattgodbolt**
## Summary
- Implement HFE v3 format exporting functionality with full specification compliance
- Refactor HFE implementation into a separate module for better code organization
- Add proper handling of weak pulses and v3 opcode collision detection
- Fix sector data reading to correctly exclude CRC bytes
- Optimize track length handling for perfect round-trip conversions

## Technical Details
- Created dedicated disc-hfe.js module with comprehensive HFE import/export functionality
- Implemented toHfe() function supporting both single and double-sided discs
- Added convertTrackToHfeV3() function handling special cases:
  - Weak pulses (value 0) replaced with HFE v3 RAND opcode
  - Potential v3 opcode collisions detected and resolved
- Used track-specific lengths in the LUT for accurate data preservation
- Added named constants for all magic numbers

## Testing
- Comprehensive test suite verifies correct HFE export and import
- Tests confirm perfect round-trip conversion of FM and MFM formatted discs
- Edge cases tested including weak pulses and potential opcode collisions
- Verified compatibility with both single and double-sided disc formats

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: tests/unit/test-disc.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { describe, it } from "vitest";
 import assert from "assert";
 
-import { Disc, DiscConfig, IbmDiscFormat, loadHfe, loadSsd, loadAdf, toSsdOrDsd } from "../../src/disc.js";
+import { Disc, DiscConfig, IbmDiscFormat, loadSsd, loadAdf, toSsdOrDsd } from "../../src/disc.js";
```
[the remark this answers is not in the document]
Comment (mattgodbolt) [reply tests/unit/test-disc.js:4]: er, it totally is, on line 193
